perun-core/src/main/java/cz/metacentrum/perun/core/impl/Utils.java

```diff
@@ -137,6 +145,68 @@ public static String join(Iterable<?> collection, String separator) {
 		return oBuilder.toString();
 	}
 
+	/**
+	 * Returns additionalUserExtSource according to the attrName. It's used for synchronization from different ExtSources. subjectFromLoginSource was obtained from the login ExtSource.
+	 *
+	 * @param sess perun session
+	 * @param attrName the key for UserExtSource in subjectFromLoginSource map
+	 * @param subjectFromLoginSource map with the subject
+	 * @return UserExtSource if exists, null if parameter attrName or subjectFromLoginSource is null or attrName doesn't start with ExtSourcesManagerImpl.USEREXTSOURCEMAPPING
+	 * @throws InternalErrorException
+	 */
+	public static UserExtSource getAdditionalUserExtSource(PerunSession sess, String attrName, Map<String, String> subjectFromLoginSource) throws InternalErrorException {
```

Comment:
Please change the method so it extracts and returns all additional user extSources from the subjectFromLoginSource. It will be maybe less effective but there will be no need to traverse through the Map each time we want to use this method, so the code will be cleaner.
Change also the name of the method to something like extractAdditionalUserExtSourcesFromSubject, because get method should just return something from an object but this method creates objects and returns them.